src/transformers/models/whisper/generation_whisper.py

```diff
@@ -733,7 +788,33 @@ def generate(
             if (prompt_ids is not None and generation_config.prompt_condition_type == "first-segment")
             else current_segments
         )
-        sequences = _pad_to_max_length(final_segments, generation_config.pad_token_id, padding="right")
+        sequences = _pad_to_max_length(
+            final_segments, 
+            pad_token_id = generation_config.pad_token_id,
+            padding="right"
+        )
+
+        if is_shortform: 
```

Comment:
Should we handle the case where `is_shortform` and `return_segments=True`? Here, I would return the same output as long-form with `return_segments=True`